exasol_advanced_analytics_framework/event_handler/context/top_level_event_handler_context.py

```diff
@@ -0,0 +1,219 @@
+        for object_proxy in list(self._valid_object_proxies):
+            self._release_object(object_proxy)
+        self._invalidate()
+
+    def _get_counter_value(self) -> int:
+        self._check_if_valid()
+        self.__counter += 1
+        return self.__counter
+
+    def _get_temporary_table_name(self) -> TableName:
+        self._check_if_valid()
+        temporary_name = self._get_temporary_db_object_name()
+        temporary_table_name = TableName(table_name=temporary_name,
+                                         schema=SchemaName(schema_name=self._temporary_schema_name))
+        return temporary_table_name
+
+    def _get_temporary_db_object_name(self) -> str:
+        temporary_name = f"{self._temporary_db_object_name_prefix}_{self._get_counter_value()}"
+        return temporary_name
+
+    def _own_object(self, object_proxy: ObjectProxy):
+        self._register_object(object_proxy)
+        self._owned_object_proxies.add(object_proxy)
+
+    def get_temporary_table(self) -> TableProxy:
+        self._check_if_valid()
+        temporary_table_name = self._get_temporary_table_name()
+        object_proxy = TableProxy(temporary_table_name)
+        self._own_object(object_proxy)
+        return object_proxy
+
+    def get_temporary_view(self) -> ViewProxy:
+        self._check_if_valid()
+        temporary_table_name = self._get_temporary_table_name()
+        object_proxy = ViewProxy(temporary_table_name)
+        self._own_object(object_proxy)
+        return object_proxy
+
+    def get_temporary_bucketfs_file(self) -> BucketFSLocationProxy:
+        self._check_if_valid()
+        temporary_path = self.get_temporary_path()
+        child_bucketfs_location = self._temporary_bucketfs_location.joinpath(temporary_path)
+        object_proxy = BucketFSLocationProxy(child_bucketfs_location)
+        self._own_object(object_proxy)
+        return object_proxy
+
+    def get_temporary_path(self):
+        temporary_path = f"{self._get_counter_value()}"
+        return temporary_path
+
+    def get_child_event_handler_context(self) -> ScopeEventHandlerContext:
+        self._check_if_valid()
+        temporary_path = self.get_temporary_path()
+        new_temporary_bucketfs_location = self._temporary_bucketfs_location.joinpath(temporary_path)
+        child_event_handler_conext = _ChildEventHandlerContext(
+            self,
+            new_temporary_bucketfs_location,
+            self._get_temporary_db_object_name(),
+            self._temporary_schema_name
+        )
+        self._child_event_handler_context_list.append(child_event_handler_conext)
+        return child_event_handler_conext
+
+    def _is_child(self, scope_event_handler_context: ScopeEventHandlerContext):
+        result = isinstance(scope_event_handler_context, _ChildEventHandlerContext) and \
+                 scope_event_handler_context._parent == self
+        return result
+
+    def _transfer_object_to(self, object_proxy: ObjectProxy,
```

Comment:
when do you need to transfer object to another context ?

Reply:
In theory, we can build something like a function graph out of event handler, for example, you have a parent function, which first calls child function 1 and then child function 2. In our framework, each function would get a EventHandlerContext. If now assume child function 1 create temporary objects and returns one of them to the parent. Then, we want that all temporary objects except the returned one, are removed after child function 1 ended. This means the return temporary object needs to be transfered from the EventHandlerContext of child function1 which we will release, to another one which won't be released. This means we can either transfer it the the EventHandlerContext of the parent function. If the returned temporary object should get the input for child function 2, we can also transfer it the EventContextHandler of child function 2.

Reply:
I will add a documentation

Reply:
thanks